pkg/controller/infinispan/resources/service/service_controller.go

```diff
@@ -209,6 +208,61 @@ func (s serviceResource) cleanupExternalExpose(excludeKind string) error {
 	return nil
 }
 
+func (s serviceResource) reconcileServiceMonitor(service *corev1.Service) (reconcile.Result, error) {
+	if s.infinispan.HasServiceMonitor() && isTypeSupported(consts.ServiceMonitorType) {
```

Comment:
We can simplify this by starting the function with:

```go
if !isTypeSupported(consts.ServiceMonitorType) {
    return reconcile.Result{}, nil
}

if s.infinispan.HasServiceMonitor() {

} else {

}
return reconcile.Result{}, nil
```

Reply:
Yes, could be easy to define. Will fix